enterprise/models.py

```diff
@@ -1823,27 +1855,49 @@ def __repr__(self):
         """
         return self.__str__()
 
-    @property
-    def enterprise_customer_uuid(self):
-        """Get the enterprise customer uuid linked to the user."""
-        try:
-            enterprise_user = EnterpriseCustomerUser.objects.get(user_id=self.user.id)
-        except ObjectDoesNotExist:
-            LOGGER.warning(
-                'User {} has a SystemWideEnterpriseRole assignment but is not linked to an enterprise!'.format(
-                    self.user.id
-                ))
-            return None
-        except MultipleObjectsReturned:
-            LOGGER.warning(
-                'User {} is linked to multiple enterprises, which is not yet supported!'.format(self.user.id)
-            )
-            return None
 
-        return enterprise_user.enterprise_customer.uuid
+@python_2_unicode_compatible
+class EnterpriseFeatureRole(UserRole):
+    """
+    Enterprise-specific feature role definitions.
 
-    def get_context(self):
+    .. no_pii:
+    """
+
+    def __str__(self):
         """
-        Return the context for this role assignment class.
+        Return human-readable string representation.
         """
-        return self.enterprise_customer_uuid
+        return "<EnterpriseFeatureRole {role}>".format(role=self.name)
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
+
+
+@python_2_unicode_compatible
+class EnterpriseFeatureUserRoleAssignment(UserRoleAssignment, EnterpriseRoleAssignmentContextMixin):
```

Comment:
@brittneyexline  One more thing. `EnterpriseRoleAssignmentContextMixin` should come before `UserRoleAssignment`, otherwise according to MRO `UserRoleAssignment.get_context` will be called every time instead of `EnterpriseRoleAssignmentContextMixin.get_context`